.github/workflows/staticcheck.yml

```diff
@@ -0,0 +1,19 @@
+name: "staticcheck"
+on: ["push", "pull_request"]
```

Comment:
Looks like this is causing it to run twice for our normal workflow, maybe we just need `push` like our other CI workflows (https://github.com/hashicorp/boundary/blob/main/.github/workflows/build.yml#L3-L6)?